File: Linguagem de Programação I/Extras/Beecrowd_Extra/1025.rb
marmores = []
T = 1
while true
    N, Q = gets.split.map(&:to_i)
    break if N == 0 && Q == 0

    marmores = Array.new(N)
    for i in 0...N
        marmores[i] = gets.to_i
    end

    marmores.sort!

    puts "CASE# #{T}:"
    T += 1
    for _ in 0...Q
        consulta = gets.to_i

        index = marmores.bsearch_index { |x| x >= consulta }

        if index.nil? || marmores[index] != consulta
            puts "#{consulta} not found"
        else
            puts "#{consulta} found at #{index + 1}"
        end
    end
end

File: Linguagem de Programação I/Extras/Beecrowd_Extra/1025.ts
import * as readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input: string[] = [];
rl.on('line', (line: string) => {
    input.push(line);
});

rl.on('close', () => {
    let T = 1;
    let line = 0;
    while (true) {
        let [N, Q] = input[line++].split(' ').map(Number);
        if (N === 0 && Q === 0) break;

        let marmores: number[] = input.slice(line, line + N).map(Number);
        line += N;
        marmores.sort((a, b) => a - b);

        console.log(`CASE# ${T++}:`);

        for (let i = 0; i < Q; ++i) {
            let consulta = Number(input[line++]);

            let index = marmores.indexOf(consulta);

            if (index === -1)
                console.log(`${consulta} not found`);
            else
                console.log(`${consulta} found at ${index + 1}`);
        }
    }
});